### PR TITLE
Fixes setting only finalized async migration as executed

### PR DESCRIPTION
**Pull request by fisehara**
Change-type: patch

### CODE REVIEW

File: src/passport-pinejs/passport-pinejs.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ const setup: ConfigLoader.SetupFunction = async (app: Express.Application) => {
 		passport.use(new LocalStrategy(checkPassword));
 
 		login = (fn) => (req, res, next) =>
-			passport.authenticate('local', (err, user?) => {
+			passport.authenticate('local', (err: any, user?: any) => {
```
Comment (Page-) [inline src/passport-pinejs/passport-pinejs.ts:57]: Why this change?

File: src/migrator/async.ts [hunk 2]
```diff
@@ -64,19 +63,10 @@ const $run = async (
...
-			'First time model has executed, skipping async migrations',
-		);
+	const executedMigrations = await getExecutedMigrations(setupTx, modelName);
 
-		return await setExecutedMigrations(setupTx, modelName, [
-			...executedMigrations,
-			...Object.keys(migrations),
-		]);
-	}
+	// if the model is new, the sync migration parts (marked by finalize=true) are already marked in the sync migration runner.
```
Comment (Page-) [inline src/migrator/async.ts:69]: And we know that it'll definitely mark them as executed before we reach this branch?

Comment (fisehara) [reply src/migrator/async.ts:69]: Yes definitively, the sync migrations are executed guaranteed before the async migrations. Any async migration that is finalized is executed in the sync migration block and will be marked as executed there.

File: src/passport-pinejs/passport-pinejs.ts (hunk 1 above)
Comment (fisehara) [reply src/passport-pinejs/passport-pinejs.ts:57]: TSC complains.

Comment (fisehara) [reply src/passport-pinejs/passport-pinejs.ts:57]: I'll remove it from this PR and make another explicitly for this TSC complains
